client/src/utils/saveAsPDF.ts

```diff
@@ -0,0 +1,24 @@
+import { Election } from '@votingworks/ballot-encoder'
+
+export default async function saveAsPDF(
+  fileNamePrefix: string,
+  election: Election,
+  precinctName?: string
+): Promise<void> {
+  const precinctNameInFileName = precinctName || 'all-precincts'
+  const data = await window.kiosk!.printToPDF()
+  const fileWriter = await window.kiosk!.saveAs({
+    defaultPath: `${`${fileNamePrefix}-${election.county.name}-${election.title}`
+      .replace(/[^a-z0-9]+/gi, '-')
+      .replace(/(^-|-$)+/g, '')
+      .toLocaleLowerCase()}-${precinctNameInFileName}.pdf`,
```

Comment:
While we're changing things, we should move the `.replace(…).replace(…).toLocaleLowerCase(…)` chain to the outside so that the precinct name is dasherized and lowercased too, right?